eng/pipelines/ci-go.yml

```diff
@@ -30,16 +30,24 @@ extends:
       - stage: "AutoRest_Go_CI"
         jobs:
           - job: "AutoRest_Go_CI"
-            displayName: "Run AutoRest CI Checks"
+            displayName: "Run AutoRest CI Checks - $(GoVersion)"
```

Comment:
Is it possible for `$(GoVersion)` to resolve in the job names in azdo?

Reply:
Oh, whoops! I was assuming this would do it, but maybe not. Let me see if I can figure this out.

We have the same challenge with the updates to the matrix files/env usage in azure-sdk-for-go, but that warrants a different solution.

Reply:
No worries, it's not urgent so if it's gonna burn a bunch of time feel free to skip it (I wasn't sure if this was just a bug or a limitation on some azdo stuff).